bundles/org.openhab.core.semantics/src/main/java/org/openhab/core/semantics/internal/SemanticsServiceImpl.java

```diff
@@ -112,4 +123,59 @@ private Predicate<? super Item> hasSynonym(String labelOrSynonym) {
             return false;
         };
     }
+
+    @Override
+    public @Nullable Class<? extends Tag> getByLabel(String tagLabel, Locale locale) {
+        Optional<SemanticTag> tag = semanticTagRegistry.getAll().stream()
+                .filter(t -> t.localized(locale).getLabel().equalsIgnoreCase(tagLabel))
+                .sorted((element1, element2) -> element1.getUID().compareTo(element2.getUID())).findFirst();
+        return tag.isPresent() ? SemanticTags.getById(tag.get().getUID()) : null;
+    }
+
+    @Override
+    public List<Class<? extends Tag>> getByLabelOrSynonym(String tagLabelOrSynonym, Locale locale) {
+        List<SemanticTag> tags = semanticTagRegistry.getAll().stream()
+                .filter(t -> getLabelAndSynonyms(t, locale).contains(tagLabelOrSynonym.toLowerCase(locale)))
+                .sorted((element1, element2) -> element1.getUID().compareTo(element2.getUID()))
+                .collect(Collectors.toList());
+        List<Class<? extends Tag>> tagList = new ArrayList<>();
+        tags.forEach(t -> {
+            Class<? extends Tag> tag = SemanticTags.getById(t.getUID());
```

Comment:
I think the Tag classes are not serving any major purpose any more, you can simply rely on `SemanticTag` instances all over this place and related ones.

Reply:
At least some of them here are for compatibility reasons with HABot.
But I will remove those not necessary (not used).